File: androidApp/features/profile/src/main/kotlin/com/cineverse/android/features/profile/ProfileScreen.kt
package com.cineverse.android.features.profile

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AccountCircle
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Person
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.SnackbarHostState
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.cineverse.core.designsystem.components.CineVerseButton
import com.cineverse.core.designsystem.components.CineVerseTextField
import com.cineverse.presentation.profile.ProfileEffect
import com.cineverse.presentation.profile.ProfileIntent
import com.cineverse.presentation.profile.ProfileViewModel
import com.cineverse.core.analytics.ThemeSettings
import org.koin.compose.koinInject

@Composable
fun ProfileScreen(
    snackbarHostState: SnackbarHostState,
    modifier: Modifier = Modifier,
    viewModel: ProfileViewModel = koinInject()
) {
    val state by viewModel.state.collectAsState()

    LaunchedEffect(Unit) {
        viewModel.effect.collect { effect ->
            when (effect) {
                is ProfileEffect.AuthSuccess -> {
                    snackbarHostState.showSnackbar("Welcome to CineVerse!")
                }
                is ProfileEffect.AuthError -> {
                    snackbarHostState.showSnackbar(effect.message)
                }
            }
        }
    }

    Column(
        modifier = modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background)
            .padding(16.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        state.user?.let { user ->
            Column(
                modifier = Modifier.fillMaxSize(),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Spacer(modifier = Modifier.height(24.dp))

                Box(
                    modifier = Modifier
                        .size(100.dp)
                        .clip(CircleShape)
                        .background(MaterialTheme.colorScheme.primary.copy(alpha = 0.1f)),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(
                        imageVector = Icons.Default.AccountCircle,
                        contentDescription = "Avatar",
                        modifier = Modifier.size(84.dp),
                        tint = MaterialTheme.colorScheme.primary
                    )
                }

                Spacer(modifier = Modifier.height(16.dp))

                Text(
                    text = user.username,
                    style = MaterialTheme.typography.headlineLarge,
                    fontWeight = FontWeight.Bold
                )
                Text(
                    text = "Premium CineVerse Member",
                    style = MaterialTheme.typography.labelSmall,
                    color = MaterialTheme.colorScheme.primary
                )

                Spacer(modifier = Modifier.height(32.dp))

                state.stats?.let { stats ->
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(16.dp)
                    ) {
                        
                        Card(
                            modifier = Modifier.weight(1f),
                            shape = RoundedCornerShape(16.dp),
                            colors = CardDefaults.cardColors(
                                containerColor = MaterialTheme.colorScheme.surface
                            )
                        ) {
                            Column(modifier = Modifier.padding(16.dp)) {
                                Text(
                                    text = "Movies Watched",
                                    style = MaterialTheme.typography.labelSmall,
                                    color = MaterialTheme.colorScheme.onSurfaceVariant
                                )
                                Spacer(modifier = Modifier.height(8.dp))
                                Text(
                                    text = "${stats.moviesWatched}",
                                    style = MaterialTheme.typography.headlineLarge.copy(fontSize = 32.sp),
                                    fontWeight = FontWeight.Bold,
                                    color = MaterialTheme.colorScheme.primary
                                )
                            }
                        }

                        Card(
                            modifier = Modifier.weight(1f),
                            shape = RoundedCornerShape(16.dp),
                            colors = CardDefaults.cardColors(
                                containerColor = MaterialTheme.colorScheme.surface
                            )
                        ) {
                            Column(modifier = Modifier.padding(16.dp)) {
                                Text(
                                    text = "Favorite Genres",
                                    style = MaterialTheme.typography.labelSmall,
                                    color = MaterialTheme.colorScheme.onSurfaceVariant
                                )
                                Spacer(modifier = Modifier.height(8.dp))
                                Text(
                                    text = stats.favoriteGenres.joinToString(separator = "\n"),
                                    style = MaterialTheme.typography.bodyMedium.copy(fontWeight = FontWeight.Bold),
                                    maxLines = 2
                                )
                            }
                        }
                    }
                }

                Card(
                    modifier = Modifier.fillMaxWidth().padding(vertical = 16.dp),
                    shape = RoundedCornerShape(16.dp),
                    colors = CardDefaults.cardColors(
                        containerColor = MaterialTheme.colorScheme.surface
                    )
                ) {
                    Row(
                        modifier = Modifier.fillMaxWidth().padding(16.dp),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Text(
                            text = "Dark Theme",
                            style = MaterialTheme.typography.bodyMedium.copy(fontWeight = FontWeight.Bold),
                            color = MaterialTheme.colorScheme.onSurface
                        )
                        val isDarkTheme by ThemeSettings.isDarkThemeFlow.collectAsState()
                        androidx.compose.material3.Switch(
                            checked = isDarkTheme,
                            onCheckedChange = { ThemeSettings.setDarkTheme(it) },
                            colors = androidx.compose.material3.SwitchDefaults.colors(
                                checkedThumbColor = MaterialTheme.colorScheme.primary,
                                checkedTrackColor = MaterialTheme.colorScheme.primary.copy(alpha = 0.5f)
                            )
                        )
                    }
                }

                Spacer(modifier = Modifier.weight(1f))

                CineVerseButton(
                    text = "Log Out",
                    onClick = { viewModel.sendIntent(ProfileIntent.Logout) },
                    modifier = Modifier.fillMaxWidth()
                )
                Spacer(modifier = Modifier.height(16.dp))
            }
        }
    }
}

File: iosApp/Screens/ProfileScreen.swift
import SwiftUI
import shared

struct ProfileScreen: View {
    @State private var viewModel = KoinHelperSwift.shared.getProfileViewModel()
    @State private var state = ProfileState(
        user: nil,
        isAuthorized: false,
        stats: nil,
        usernameInput: "",
        passwordInput: "",
        isLoading: false,
        error: nil
    )

    @State private var alertMessage: String?
    @State private var showAlert: Bool = false

    var body: some View {
        NavigationView {
            ZStack {
                AppColors.background.edgesIgnoringSafeArea(.all)
                ProfileDetailsLayout(state: state, onLogout: {
                    viewModel.sendIntent(intent: ProfileIntent.Logout())
                })
            }
            .navigationTitle("My Profile")
        }
        .task {
            for await currentState in viewModel.state {
                if let currentState = currentState {
                    self.state = currentState
                }
            }
        }
    }
}

struct ProfileDetailsLayout: View {
    let state: ProfileState
    let onLogout: () -> Void

    var body: some View {
        VStack(spacing: 24) {
            Spacer().frame(height: 10)

            ZStack {
                Circle()
                    .fill(AppColors.primary.opacity(0.1))
                    .frame(width: 100, height: 100)

                Image(systemName: "person.crop.circle.fill")
                    .resizable()
                    .aspectRatio(contentMode: .fit)
                    .frame(width: 80, height: 80)
                    .foregroundColor(AppColors.primary)
            }

            if let user = state.user {
                VStack(spacing: 4) {
                    Text(user.username)
                        .font(AppTypography.headingLarge)
                    Text("Premium CineVerse Member")
                        .font(AppTypography.labelSmall)
                        .foregroundColor(AppColors.primary)
                }
            }

            if let stats = state.stats {
                HStack(spacing: 16) {

                    VStack(alignment: .leading, spacing: 8) {
                        Text("Movies Watched")
                            .font(AppTypography.labelSmall)
                            .foregroundColor(.gray)

                        Text("\(stats.moviesWatched)")
                            .font(.system(size: 32, weight: .bold))
                            .foregroundColor(AppColors.primary)
                    }
                    .padding()
                    .frame(maxWidth: .infinity, alignment: .leading)
                    .background(AppColors.surface)
                    .cornerRadius(16)

                    VStack(alignment: .leading, spacing: 8) {
                        Text("Favorite Genres")
                            .font(AppTypography.labelSmall)
                            .foregroundColor(.gray)

                        Text(stats.favoriteGenres.joined(separator: "\n"))
                            .font(AppTypography.bodyMedium)
                            .fontWeight(.bold)
                            .foregroundColor(.primary)
                            .lineLimit(2)
                    }
                    .padding()
                    .frame(maxWidth: .infinity, alignment: .leading)
                    .background(AppColors.surface)
                    .cornerRadius(16)
                }
                .padding(.horizontal)
            }

            HStack {
                Text("Dark Theme")
                    .font(AppTypography.bodyMedium)
                    .fontWeight(.bold)
                    .foregroundColor(.primary)
                Spacer()
                Toggle("", isOn: Binding<Bool>(
                    get: { ThemeSettings.shared.isDarkTheme() },
                    set: { ThemeSettings.shared.setDarkTheme(enabled: $0) }
                ))
                .labelsHidden()
            }
            .padding()
            .background(AppColors.surface)
            .cornerRadius(16)
            .padding(.horizontal)

            Spacer()

            CineVerseButton(text: "Log Out") {
                onLogout()
            }
            .padding(.horizontal)
            .padding(.bottom, 24)
        }
    }
}
